mutation-testing/check-packages-and-shards/action.yml

```diff
@@ -0,0 +1,123 @@
+name: Check Packages and Shards
+
+outputs:
+  run_big_packages:
+    value: ${{ steps.check_packages_and_shards.outputs.run_big_packages }}
+  big_packages_with_shards:
+    value: ${{ steps.check_packages_and_shards.outputs.big_packages_with_shards }}
+  run_small_packages:
+    value: ${{ steps.check_packages_and_shards.outputs.run_small_packages }}
+  small_packages_with_shards:
+    value: ${{ steps.check_packages_and_shards.outputs.small_packages_with_shards }}
+
+runs:
+  using: "composite"
+
+  steps:
+    - name: Checkout stacks-core repo
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+
+    - name: Install cargo-mutants
+      shell: bash
+      run: cargo install --version 23.12.2 cargo-mutants
+
+    - name: Relative diff
+      shell: bash
+      run: |
+        git diff origin/${{ github.base_ref }}.. > git.diff
+
+    - name: Update git diff
+      shell: bash
+      run: |
+        input_file="git.diff"
+        temp_file="temp_diff_file.diff"
+
+        # Check if the commands exist on the host
+        for cmd in tac awk sed; do
+          command -v "${cmd}" > /dev/null 2>&1 || echo "Missing command: ${cmd}"
+        done
+
+        # Reverse the file, remove 4 lines after '+++ /dev/null', then reverse it back (editors can't go backwards - to remove lines above)
+        tac "$input_file" > "$temp_file" && mv "$temp_file" "$input_file"
+        sed '/+++ \/dev\/null/{n;N;N;N;d;}' "$input_file" > "$temp_file" && mv "$temp_file" "$input_file"
+        tac "$input_file" > "$temp_file" && mv "$temp_file" "$input_file"
+
+        # Remove the lines between '+++ /dev/null' (included) and 'diff --git a/'
+        awk '
+          BEGIN { in_block=0 }
+          /\+\+\+ \/dev\/null/ { in_block=1; next }
+          in_block && /diff --git a\// { in_block=0; print; next }
+          !in_block
+        ' "$input_file" > "$temp_file" && mv "$temp_file" "$input_file"
```

Comment:
i would break that `&&` into a separate line, with a check that the file exists. 
```
...
' "$input_file" > "$temp_file"
if [ -f "$temp_file" ]; then
  mv "$temp_file" "$input_file"
else
  echo "missing file"
   exit 1
fi
```